library/core/src/num/nonzero.rs

```diff
@@ -513,6 +513,33 @@ nonzero_unsigned_operations! {
 macro_rules! nonzero_signed_operations {
     ( $( $Ty: ident($Int: ty) -> $Uty: ident($Uint: ty); )+ ) => {
         $(
+            #[stable(feature = "nonzero_checked_ops", since = "1.63.0")]
+            impl Neg for $Ty {
+                type Output = Self;
+                /// Negate the non-zero value.
+                ///
+                /// # Example
+                ///
+                /// ```
+                #[doc = concat!("# use std::num::", stringify!($Ty), ";")]
+                ///
+                /// # fn main() { test().unwrap(); }
+                /// # fn test() -> Option<()> {
+                #[doc = concat!("let pos = ", stringify!($Ty), "::new(1)?;")]
+                #[doc = concat!("let neg = ", stringify!($Ty), "::new(-1)?;")]
+                ///
+                /// assert_eq!(-pos, neg);
+                /// assert_eq!(-neg, pos);
+                /// # Some(())
+                /// # }
+                /// ```
```

Comment:
Just curious: why all of this function shenanigans in the docktest?  Why not simply have:

```rust
/// ```
#[doc = concat!("# use std::num::", stringify($Ty), ";")]
#[doc = concat!("let pos = ", stringify!($Ty), "::new(1)?;")]
#[doc = concat!("let neg = ", stringify!($Ty), "::new(-1)?;")]
/// assert_eq!(-pos, neg);
/// assert_eq!(-neg, pos);
/// ```
```

Reply:
@eggyal Ah okay, I'm getting it now. Your proposition does not work because of the `?` unwrappings that make the return type of the hidden `main` function non-trivial :\\
I'm happy to find a better solution if anyone has, though.

Reply:
Ah, yes, I missed those.  But you could just `.unwrap()` instead of `?`, no?

Reply:
This does work indeed. Maybe I valued docs readability over sources simplicity? Not sure what's the best tradeoff here.. are there precedents in the docs already?

Reply:
It's not something for this PR anyway, but I think it's more common to see `.unwrap()` in the doctest.  Not a big deal to me either way!  Thanks for explaining :)

Reply:
Yeah, I was scrolling elsewhere in the docs and figured that `.uwrap()` was pretty common idiom indeed. Now I'm pretty much convinced that I did this just because I found the examples nicer with `?` instead. I'll leave it as-is unless someone has another reason not to use them :)